library-conan/conanfile.py

```diff
@@ -0,0 +1,69 @@
+import os
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.scm import Git
+from conan.tools.files import update_conandata
+
+class DPPConan(ConanFile):
+    name = "dpp"
+    version = "10.0.29"
+    default_user = "brainboxdotcc"
+    default_channel = "testing"
+    license = "https://github.com/brainboxdotcc/DPP/blob/master/LICENSE"
+    url = "https://github.com/brainboxdotcc/DPP"
+    description = "D++ is a lightweight and efficient library for Discord"
+    topics = ("discord")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    def build_requirements(self):
+        self.requires("nlohmann_json/3.11.2", headers=True, libs=True)
+        self.requires("openssl/3.1.2", headers=True, libs=True)
+        self.requires("zlib/1.3", headers=True, libs=True)
+        self.requires("libsodium/cci.20220430", headers=True, libs=True)
+        self.requires("opus/1.4", headers=True, libs=True)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self)
+
+    def export(self):
+        git = Git(self, os.path.dirname(self.recipe_folder))
+        scm_url, scm_commit = git.get_url_and_commit()
+        update_conandata(self, {"sources": {"commit": scm_commit, "url": scm_url}})
+
+    def source(self):
+        git = Git(self)
+        sources = self.conan_data["sources"]
+        git.clone(url=sources["url"], target=".")
+        git.checkout(commit=sources["commit"])
+
+    def generate(self):
+        deps = CMakeDeps(self)
+        deps.generate()
+        tc = CMakeToolchain(self)
+        tc.cache_variables["CONAN_EXPORTED"] = True
+        tc.cache_variables["BUILD_VOICE_SUPPORT"] = True
+        tc.cache_variables["DPP_BUILD_TEST"] = False
+        tc.cache_variables["BUILD_SHARED_LIBS"] = False
+        tc.generate()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def package(self):
+        cmake = CMake(self)
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = ["dpp"]
```

Comment:
As commented in https://github.com/conan-io/conan/issues/15985#issuecomment-2027752183 this might need more info, as the folders generated in ``cmake.install()`` are not matching the defaults.